app/views/layouts/application.html.erb

```diff
@@ -11,24 +11,15 @@
   <body>
 
     <nav class = "topnav">
-      <%= link_to "Home", "/" %>
-      <%= link_to "All Merchants", "/merchants"%>
-      <%= link_to "All Items", "/items"%>
-      <%= link_to "Cart: #{cart.total_items}", "/cart" unless current_admin? %>
-
       <% if current_user == nil %>
-        <%= link_to "Register", "/register" %>
-        <%= link_to "Log in", "/login" %>
+        <%= render partial: './users/visitor_nav' %>
+      <% elsif current_user.role == "regular" %>
+        <%= render partial: './users/user_nav' %>
       <% elsif current_user.admin? %>
-        <%= link_to "All Users", "/admin/users" %>
-        <%= link_to "Dashboard", "/admin/dashboard" %>
+        <%= render partial: './users/admin_nav' %>
       <% elsif current_user.merchant? %>
-        <%= link_to "Dashboard", "/merchant/dashboard" %>
+        <%= render partial: './users/merchant_nav' %>
```

Comment:
If you want to, we could clean this up even more with guard statements - we'd end up with only four lines in our navbar!

Reply:
Like put these statements ```<% elsif current_user.admin? %>``` into the partial itself?

Reply:
No they would remain in this view and be like this:
```
<%= render partial: './users/user_nav' if current_user.regular? %>
<%= render partial: './users/admin_nav' if current_user.admin? %>
```
... and so on